src/components/controllers/tutorial/index.js

```diff
@@ -158,44 +180,52 @@ export default class Tutorial extends Component {
 
 function TutorialView({
 	step,
-	route,
 	loading,
 	code,
 	error,
 	Runner,
 	CodeEditor,
 	clearError
 }) {
-	const content = useRef();
+	const content = useRef(null);
 
 	const tutorial = useContext(TutorialContext);
 
 	const [showCodeOverride, toggleCode] = useReducer(s => !s, true);
 
-	const { lang, solved } = useStore(['lang', 'solved']).state;
-	const fullPath = route.path.replace(':step?', step || route.first);
-	const page = usePage({ path: fullPath }, lang);
-	const title = (page && page.meta.title) || route.title;
-	const solvable = page && page.meta.solvable === true;
-	const hasCode = page && page.meta.code !== false && step && step !== 'index';
+	const { url } = useLocation();
+	const { params } = useRoute();
+	const [lang] = useLanguage();
+	const { solved } = useContext(SolutionContext);
+
+	const { html, meta } = useContent([
+		lang,
+		!params.step ? 'tutorial/index' : url
+	]);
+	useTitle(meta.title);
+	useDescription(meta.description);
+
+	const solvable = meta.solvable === true;
+	const hasCode = meta.code !== false && step && step !== 'index';
 	const showCode = showCodeOverride && hasCode;
-	loading =
-		!page.html || (showCode && (!!page.loading || !Runner || !CodeEditor));
-	const initialLoad = !page.html || !Runner || !CodeEditor;
+	loading = !html || (showCode && (!Runner || !CodeEditor));
+	const initialLoad = !html || !Runner || !CodeEditor;
 
 	// Scroll to the top after loading
 	useEffect(() => {
 		if (!loading && !initialLoad) {
 			content.current.scrollTo(0, 0);
 		}
-	}, [fullPath, loading, initialLoad]);
+	}, [url, loading, initialLoad]);
 
 	// Preload the next chapter
-	useEffect(() => {
-		if (page.meta && page.meta.next) {
-			getContent([lang, page.meta.next]);
-		}
-	}, [page.meta && page.meta.next, fullPath]);
+	// TODO: Webpack creates a circular dependency that
+	// it cannot resolve. Temporarily disabled
+	//useEffect(() => {
+	//	if (meta && meta.next) {
+	//		getContent([lang, meta.next]);
+	//	}
+	//}, [meta && meta.next, url]);
```

Comment:
As the comment says, unfortunately lazy loading the tutorial creates a circular dep that Webpack is not a big fan of. I couldn't figure out any way to stop it from dying.

Suggestions certainly welcome if you got 'em.